### ISSUE TITLE
[Dialog] The dialog component doesn't have the new a11y updated Modal styling

### ISSUE THREAD

**Issue by alina-jacob** [carbon-design-system/carbon#21923]
## Acceptance criteria

The dialog component doesn't have the new a11y updated Modal styling in React and WC.
- [Design spec](https://github.com/carbon-design-system/carbon/issues/20796)

[Current implementation](https://react.carbondesignsystem.com/?path=/story/preview-preview-dialog--modal)
<img width="1725" height="713" alt="Image" src="https://github.com/user-attachments/assets/2abb2a3e-50bc-4e67-bbc2-c1e0b9c8935b" />

[Expected implementation](https://react.carbondesignsystem.com/?path=/story/components-modal--default&globals=theme:g100)
<img width="1727" height="667" alt="Image" src="https://github.com/user-attachments/assets/a47bcc89-c800-4af6-8410-dee4b8b2a45f" />
